Tutorials/Python_Developer_Tutorials/foundation/Commanding-Local/Proxy-Local-Commands-Tutorial_.py

```diff
@@ -1,87 +1,112 @@
+command = faraday_cmd.CommandLocalGPIO(0, 0, 0, (gpioallocations.LED_1 | gpioallocations.LED_2), 0, 0)
 faraday_1.POST(local_device_callsign, local_device_node_id, faraday_1.CMD_UART_PORT, command)
 
+
 ###############
 ## ECHO MESSAGE
 ###############
-print "/n** Beginning ECHO command test** /n"
-#Use the general command library to send a text message to the Faraday UART "ECHO" command. Will only ECHO a SINGLE packet. This will send the payload of the message back (up to 62 bytes, this can be updated in firmware to 124!)
-originalmsg = "This will ECHO back on UART" #Cannot be longer than max UART payload size!
-command = faradaycommands.commandmodule.create_command_datagram(faraday_cmd.CMD_ECHO, originalmsg)
-faraday_1.POST(local_device_callsign, local_device_node_id, faraday_1.CMD_UART_PORT, command)
-
-#Retrive waiting data packet in UART Transport service number for the COMMAND application (Use GETWait() to block until ready or return False).
-rx_echo_raw = faraday_1.GETWait(local_device_callsign, local_device_node_id, faraday_1.CMD_UART_PORT, sec_timeout = 3)  # Wait for up to 3 seconds for data to arrive
-
-#Now parse data again
-b64_data = rx_echo_raw[0]['data']
-echo_decoded = faraday_1.DecodeRawPacket(b64_data)
+print "\n** Beginning ECHO command test** \n"
+originalmsg = "This will ECHO back on UART"  # Cannot be longer than max UART Transport layer payload size!
 
-#Display information
+# Display information
+print "**Sending**\n"
 print "Original Message: ", originalmsg
-print "RAW Received BASE64 ECHO'd Message:", b64_data
-print "Decoded received ECHO'd Message:", echo_decoded #Note that ECHO sends back a fixed packed regardless. Should update to send back exact length.
-
 
+#  Create command packet
+command = faradaycommands.commandmodule.create_command_datagram(faraday_cmd.CMD_ECHO, originalmsg)
+#  Send command
+faraday_1.POST(local_device_callsign, local_device_node_id, faraday_1.CMD_UART_PORT, command)
+#  Retrive waiting data packet in UART Transport service number for the COMMAND application (Use GETWait() to block until ready or timeout
+rx_echo_raw = faraday_1.GETWait(local_device_callsign, local_device_node_id, faraday_1.CMD_UART_PORT,
+                                sec_timeout=3)  # Wait for up to 3 seconds for data to arrive
+
+try:
+    # Decode BASE 64 encoded data from Proxy - Only using the [0] index packet (first) if multiple are retrieved
+    b64_data = rx_echo_raw[0]['data']
+    echo_decoded = faraday_1.DecodeRawPacket(b64_data)
+
+    # Display information
+    print "\n**Receiving**\n"
+    print "Decoded received ECHO'd Message:", echo_decoded  # Note that ECHO sends back a fixed packed regardless. Should update to send back exact length.
+    print "\nRAW Received BASE64 ECHO'd Message:", b64_data
+    print "\nDecoded BASE64 RAW Bytes:", repr(echo_decoded)
+
+except Exception as e:
+    print "Failed ECHO due to type error:", e
```

Comment:
Not really a type error. Would be MOST useful to print out the error JSON message here.